File: 11478/11478.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    11478.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/05/19 12:17:48 by jihoolee          #+#    #+#              #
#    Updated: 2022/05/19 12:29:29 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main():
    string = input()

    substr = set([string[idx:idx + size]
                    for size in range(1, len(string) + 1)
                        for idx in range(0, len(string) - size + 1)])
    print(len(substr))
if (__name__ == '__main__'):
    main()

File: 11478/11478.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   11478.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/05/19 12:05:30 by jihoolee          #+#    #+#             */
/*   Updated: 2022/05/19 12:14:52 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>
#include <set>

int main(void) {
  std::string           str;
  std::set<std::string> substr;

  std::cin >> str;
  for (int size = 1; size <= str.length(); ++size) {
    for (int idx = 0; idx + size <= str.length(); ++idx)
      substr.insert(str.substr(idx, size));
  }
  std::cout << substr.size();
  return 0;
}
